src/server.c

```diff
@@ -5453,7 +5453,7 @@ int processCommand(client *c) {
           c->cmd->proc != quitCommand &&
           c->cmd->proc != resetCommand &&
         !(c->cmd->proc == shutdownCommand &&
-          c->argc == 2 &&
+          c->argc >= 2 && // FIXME
```

Comment:
this thing became too complicated.
we do have a plan to soon promote this list of commands to command flags, which will be ok for SCRIPT KILL since it's a sub-command, but will not be ok for SHUTDOWN.
i suppose the only way out is to permit the shutdown command, and add some explicit check inside `shutdownCommand` to fail with a similar error as the one processCommand uses.